llvm/lib/Target/AMDGPU/SIMachineFunctionInfo.cpp

```diff
@@ -169,6 +169,15 @@ SIMachineFunctionInfo::SIMachineFunctionInfo(const Function &F,
     VGPRForAGPRCopy =
         AMDGPU::VGPR_32RegClass.getRegister(ST.getMaxNumVGPRs(F) - 1);
   }
+
+  if (STI->hasArchitectedSGPRs()) {
+    ArgInfo.ArchitectedWorkGroupIDX =
```

Comment:
I think it's OK for the lowering to directly consume the hardcoded register numbers